### DISCUSSION TITLE
How do you prevent setting a new src from restarting the video?

### DISCUSSION THREAD

**Discussion by pac96** [videojs/video.js#8695] [Q&A]
I have a resolution player that sets a new src when we change the resolution. However, every time the new src is loaded, the video briefly restarts from 0:00 then goes back to the current time. Is there a way to set the src without restarting the video? Is there any guide on how to load a new src for a video and have it start at the same time it was at before?

**Comment by logan858** [comment]
> I have a resolution player that sets a new src when we change the resolution. However, every time the new src is loaded, the video briefly restarts from 0:00 then goes back to the current time. Is there a way to set the src without restarting the video? Is there any guide on how to load a new src for a video and have it start at the same time it was at before?

depending what your setup, it's likely re-rendering the video when the src updates due to resolution change, unless you're going to implement some sort of setup where your video players lifecycle lives outside this, then you'll have to deal with the re-rendering.  you can pass back in your currenttime to the video player and set it again like so:

`
player.on("ready", () => {
player.currentTime(yourPreviousTime)
})
`

to avoid that flash of a split second of the video starting back at 0 seconds before adjusting to the previous time, you may need to go off of "loadedmetadata" and/or "canplaythrough" to avoid that split second flash of the player re-rendering at time: 0 seconds.  i'd also try playing around with preloading metadata in the video.js options too.

`
player.on("loadedmetadata", () => {
            if (yourPreviousTime) {
              player.currentTime(yourPreviousTime);
            }
          });
`
